shared/reducers/fs.js

```diff
@@ -204,11 +203,37 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
     case FsGen.discardEdit:
       // $FlowFixMe
       return state.removeIn(['edits', action.payload.editID])
-    case FsGen.editFailed:
-      // $FlowFixMe
-      return state.setIn(['edits', action.payload.editID, 'status'], 'failed')
     case FsGen.fsError:
-      return state.setIn(['errors', Constants.makeUUID()], action.payload.error)
+      const {erroredAction, error} = action.payload.error
+      logger.error('error (fs)', erroredAction.type, error)
+      const nextState: Types.State = state.setIn(['errors', Constants.makeUUID()], action.payload.error)
+
+      switch (erroredAction.type) {
```

Comment:
@chrisnojima I ended up moving the error stuff for download, upload, and edits here, so saga don't need to worry about what component need information about errors. It also don't require dispatching multiple actions from a single saga action handler. Please let me know if you think there's a better way to do this.